molecularnodes/io/parse/mda.py

```diff
@@ -156,13 +156,17 @@ def elements(self) -> List[str]:
             elements = self.ag.elements.tolist()
         except:
             try:
-                elements = [
-                    "BB" if x == "BB" else
-                    "SC" if x.startswith("SC") else
-                    "GL" if x.startswith("GL") else
-                    "CD" if x.startswith("D") else
-                    mda.topology.guessers.guess_atom_element(x) for x in self.ag.atoms.names
-                ]
+                elements = [mda.topology.guessers.guess_atom_element(x) for x in self.ag.atoms.names]
```

Comment:
Yeah, I guess I overlooked the potential for bugs arising from MDA.topology.guessers.guess_atom_element() method. So, here's what I'm thinking of changing this code to: 
```
elements = [
                    x if x in data.coarse_grain_particles.keys() else
                    x if x in data.elements.keys() else
                    mda.topology.guessers.guess_atom_element(x) for x in self.ag.atoms.names]
```

If the atom name is within either the `data.coarse_grain_particles` or `data.elements` dictionaries, then the code will use the atom name string as the element symbol. Otherwise, it will use the guess_atom_element() method to get an element symbol. Since the atom name field in .pdb and .cif files are usually fully capitalized, the second if statement will most likely never return True (element symbol keys in `data.elements` are not fully capitalized). 

There's still the potential for bugs since the `coarse_grain_particles` dictionary is likely to always be incomplete. I'm not a CG expert, but there are numerous models with their own naming scheme for particles.

Reply:
Oh, there's a bug already in this code. The atom name `CD` maps to a "MartiniLipidCarbonD" particle in the `data.coarse_grain_particles` dict. But, there are instances where the `CD` atom name  should map to a carbon, i.e. the delta-carbon in lysine. The code above prioritizes coarse-grained particles over atoms. 

Personally, I think erring on the side of atom names mapping directly to elements rather than accounting for ambiguous/conflicting names from coarse-grained models is safer.

Reply:
There really should be a check against residue name as well as atom name before assigning an element symbol. The residue name would give the context needed to have confidence in labeling the atom name to an element symbol.

Reply:
We can try and check against the res names, or we could potentially add a "namespace" option when importing MD trajectories. Given there are going to be so many possible weird and wonderful combinations of atom names, res names etc - it might be simpler for use to just provide an option for the user to select a dictionary to lookup from. 

Currently - we could maybe do a "standard elements" dictionary, or a "Martini forcefield" option. This would enable adding support for other force fields & naming conventions, but the onus would be on users who want to use those data types to potentially supply new dictionaries that we could then add in.

Not sure it would be best course of action (not a big fan of adding _yet more_ UI elments to fiddle with when importing) but it might solve the headache of these weird mismatches in naming between data formats.

Reply:
> Not sure it would be best course of action (not a big fan of adding _yet more_ UI elments to fiddle with when importing) but it might solve the headache of these weird mismatches in naming between data formats.

I agree with this. Adding more switches at the front, especially for such a nuanced thing, isn't great. But, the diversity of atoms/particles possible can result in bugs. This seems to be a larger, very complex issue. Maybe deserves its own Issue and thorough consideration. In regards to this issue/PR, can a message be passed to the user suggesting that they check for mislabeling of atom metadata (like `vdw_radii` and `mass` attributes)? 

As far as I can tell, there's not a great way to pass messages to users; I can't get print statements in scripts to print anywhere. Warning/error messages that get raised when using the MN gui panels are small little text boxes that disappear rather quickly, with no log of them afterwards. 

> Oh, there's a bug already in this code. The atom name `CD` maps to a "MartiniLipidCarbonD" particle in the `data.coarse_grain_particles` dict. But, there are instances where the `CD` atom name should map to a carbon, i.e. the delta-carbon in lysine. The code above prioritizes coarse-grained particles over atoms.
> 
> Personally, I think erring on the side of atom names mapping directly to elements rather than accounting for ambiguous/conflicting names from coarse-grained models is safer.

So, what's the play here? I still think the atom-type first approach is safer; CG particles are used sparingly.

Reply:
Just FYI, this is not a new problem to be having. MDAnalysis still seems to have the issue raised off and on since 2015. The most recent issue that I've found is https://github.com/MDAnalysis/mdanalysis/issues/3704 and an active PR https://github.com/MDAnalysis/mdanalysis/pull/3753, both from the same author. Digging through the recent updates to the PR, it seems like MDA 3.0 may have an improved guesser suite. Maybe implementing any sort of guesser on the MN side should wait until MDA 3.0 comes out?

Reply:
With an improved guesser on the way then we certainly shouldn't implement anything ourselves.

I think a temporary stopgap would be adding the option to choose a different dictionary to guess against - but I think that can be handled in a spearate PR and for now we just try the default dictionary first, then default to the guesser.

Reply:
Unfortunately, I can't find any official planned release date for MDA 3.0. 

For now, the code will be: 

```python
elements = [
                    x if x in data.elements.keys() else
                    mda.topology.guessers.guess_atom_element(x) for x in self.ag.atoms.names]
```

but, doing this will result in using the guesser most of the time. Keys in the `data.elements` dictionary are the element symbols, which follow Upperlower case formatting of strings. Atom names in almost all topologies are fully capitalized strings so won't pass the boolean test.